test/models/release_test.exs

```diff
@@ -52,42 +59,50 @@ defmodule HexWeb.ReleaseTest do
     ecto = HexWeb.Repo.get_by!(Package, name: "ecto")
 
     assert {:ok, _} =
-           Release.create(decimal, rel_meta(%{version: "0.1.0", app: "decimal", requirements: %{}}), "")
+           Release.create(decimal, rel_meta(%{version: "0.1.0", app: "decimal", requirements: []}), "")
+           |> HexWeb.Repo.insert
 
     assert {:ok, _} =
-           Release.create(ecto, rel_meta(%{version: "0.1.0", app: "ecto", requirements: %{"decimal" => "~> 0.1"}}), "")
-
-    assert {:error, [version: "is invalid"]} =
-           Release.create(ecto, rel_meta(%{version: "0.1", app: "ecto"}), "")
-
-    assert {:error, [requirements: [{"decimal", "invalid requirement: \"fail\""}]]} =
-           Release.create(ecto, rel_meta(%{version: "0.1.1", app: "ecto", requirements: %{"decimal" => "fail"}}), "")
-
-    assert {:error, [requirements: requirements]} =
-            Release.create(ecto, rel_meta(%{version: "0.1.1", app: "ecto", requirements: %{"decimal" => "~> 1.0"}}), "")
+           Release.create(ecto, rel_meta(%{version: "0.1.0", app: "ecto", requirements: [%{name: "decimal", requirement: "~> 0.1"}]}), "")
+           |> HexWeb.Repo.insert
+
+    assert %{version: [{"is invalid", [type: HexWeb.Version]}]} =
+           Release.create(ecto, rel_meta(%{version: "0.1", app: "ecto"}), "") |> extract_errors
+    
+    assert %{requirements: [%{requirement: [{"invalid requirement: \"~> fail\"", []}]}]} =
+           Release.create(ecto, rel_meta(%{version: "0.1.1", app: "ecto", requirements: [%{name: "decimal", requirement: "~> fail"}]}), "") |> extract_errors
+           
+    assert %{requirements: [%{requirement: [{"Conflict on decimal\n  mix.exs: ~> 1.0\n", []}]}]} =
+           Release.create(ecto, rel_meta(%{version: "0.1.1", app: "ecto", requirements: [%{name: "decimal", requirement: "~> 1.0"}]}), "") |> extract_errors
+  end
 
-    # TODO: Remove double check when Hex 0.12.0 is out
-    assert "Conflict on decimal\n  mix.exs: ~> 1.0\n" == requirements or
-           "Failed to use \"decimal\" because\n  You specified ~> 1.0 in your mix.exs\n" == requirements
```

Comment:
No worries, the next version of Hex updates the conflict resolution error messages. I will fix it later.